sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -667,7 +667,14 @@ object JdbcUtils extends Logging {
     val getConnection: () => Connection = createConnectionFactory(options)
     val batchSize = options.batchSize
     val isolationLevel = options.isolationLevel
-    df.foreachPartition(iterator => savePartition(
+    val numPartitions = options.numPartitions
+    val repartitionedDF =
+      if (numPartitions != null && numPartitions.toInt != df.rdd.getNumPartitions) {
```

Comment:
Normally, based on my understanding, users only cares the maximal number of connections. Thus, no need to repartition it when `numPartitions.toInt >= df.rdd.getNumPartitions`, right?